src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/SendEmailsForGoCardlessMandateEventsHandlerTest.java

```diff
@@ -48,43 +53,45 @@
+            .withResourceType(GoCardlessResourceType.MANDATES)
             .withLinksOrganisation(organisationIdentifier);
 
     @Test
-    public void handle_onAFailedMandateGoCardlessEvent_shouldRegisterAPayEventAsMandateFailed() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture.withAction("failed").toEntity());
+    public void assertEmailIsSentForFailedMandateGoCardlessEvent() {
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture.withAction("failed").toEntity();
 
         when(mockMandateQueryService.findByGoCardlessMandateIdAndOrganisationId(
                 goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation()))
                 .thenReturn(mandateFixture.toEntity());
 
-        goCardlessMandateHandler.handle(goCardlessEvent);
+        sendEmailsForGoCardlessEventsHandler.handle(List.of(goCardlessEvent));
 
-        verify(userNotificationService).sendMandateFailedEmailFor(mandateFixture.toEntity());
+        verify(userNotificationService, times(1)).sendMandateFailedEmailFor(mandateFixture.toEntity());
     }
 
     @Test
-    public void handle_onACancelledMandateGoCardlessEvent_shouldRegisterAPayEventAsMandateCancelled() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture.withAction("cancelled").toEntity());
+    public void assertEmailIsSentForCancelledMandateGoCardlessEvent() {
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture.withAction("cancelled").toEntity();
 
         when(mockMandateQueryService.findByGoCardlessMandateIdAndOrganisationId(
                 goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation()))
                 .thenReturn(mandateFixture.toEntity());
 
-        goCardlessMandateHandler.handle(goCardlessEvent);
+        sendEmailsForGoCardlessEventsHandler.handle(List.of(goCardlessEvent));
 
-        verify(userNotificationService).sendMandateCancelledEmailFor(mandateFixture.toEntity());
+        verify(userNotificationService, times(1)).sendMandateCancelledEmailFor(mandateFixture.toEntity());
     }
 
     @Test
-    public void handle_onCreateMandateGoCardlessEvent_shouldThrowExceptionWhenEventHasNoLinkedMandate() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture
+    public void shouldThrowExceptionWhenEventHasNoLinkedMandate() {
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture
                 .withLinksOrganisation(GoCardlessOrganisationId.valueOf("does_not_exist"))
                 .withAction("cancelled")
                 .withLinksMandate(null)
-                .toEntity());
+                .toEntity();
 
         thrown.expect(GoCardlessEventHasNoMandateIdException.class);
-        goCardlessMandateHandler.handle(goCardlessEvent);
+        sendEmailsForGoCardlessEventsHandler.handle(List.of(goCardlessEvent));
+        verify(userNotificationService, never()).sendMandateCancelledEmailFor(any());
```

Comment:
There’s also `verifyZeroInteractions(userNotificationService);` that could be used here.

Reply:
nice, `verifyZeroInteractions` is more convenient!